### PR TITLE
Add multi-tenancy support to generic OAuth providers.

### LINKED ISSUE #7990: OAuthProvider with tenant support
### [REQUIRED] Step 1: Describe your environment

  * Xcode version:  12.4
  * Firebase SDK version: 7.10.0
  * Installation method: `CocoaPods ` 
  * Firebase Component: Auth

#### Steps to reproduce:

there is a problem when we try to signin with a specific tenant and specific provider using OAuthProvider(see code below for a better understand)

in the file `FIROAuthProvider.m` line `311` where the params are added in the URL, the param `tid` is missed, and in some cases this param is needed to the correct functionality of the authentication, don't know if it's missed in the other providers, but in this case of the oauth provider is needed with a multi tenant implementation.

something to add is that Android SDK create the url in a right way when we use the tenant id in the same way.

#### Relevant Code:

let's use this code to perform the signin...
```
let auth = Auth.auth()
auth.tenantID = "tenant_id"
let provider = OAuthProvider(providerID: "provider_id", auth: auth)

auth.signIn(with: provider, uiDelegate: nil) { authResult, error in
// handle here the result
}
```

### PR DESCRIPTION
**Pull request by rosalyntan**
Fixes #7990

### CODE REVIEW

**Review by paulb777 — COMMENTED** [review]
Thanks @rosalyntan! Please add a Changelog entry.

**Review by renkelvin — APPROVED** [review]
Thanks @rosalyntan!

**Comment by rosalyntan** [comment]
> Thanks @rosalyntan! Please add a Changelog entry.

Done!

File: FirebaseAuth/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Unreleased
-- [changed] Added a `X-Firebase-GMPID` header to network requests.
+- [changed] Added a `X-Firebase-GMPID` header to network requests. (#9046)
+- [fixed] Added multi-tenancy support to generic OAuth providers. (#9100)
```
Comment (paulb777) [inline FirebaseAuth/CHANGELOG.md:3]: Recommended practice is to specify the issue number when there is both a bug and PR - #7990 in this case.

**Review by paulb777 — APPROVED** [review]

File: FirebaseAuth/CHANGELOG.md (hunk 1 above)
Comment (rosalyntan) [reply FirebaseAuth/CHANGELOG.md]: Ah, gotcha! Done.

**Pull request merged**